lib/productions/interface.js

```diff
@@ -40,4 +41,12 @@ export class Interface extends Container {
   get type() {
     return "interface";
   }
+
+  *validate(defs) {
+    if (!this.partial && this.extAttrs.every(extAttr => extAttr.name !== "Exposed")) {
+      const message = `Non-partial interfaces must have [Exposed] extended attribute.`;
```

Comment:
Will this cover namespaces too?  Or do that separate?

Reply:
This is only for interfaces, the namespace part is below in lib/productions/namespace.js 😀

Reply:
Also, we want to  make these messages as friendly as possible, so we should include a "how to fix". So...  
```suggestion
      const message = `Interfaces and namespaces must have [Exposed] extended attribute. To fix, add, for example, [Exposed=(Window)]. Please also consider carefully if your interface should also be exposed in a Worker scope. Refer to the WebIDL spec section on Exposed for more information.`;
```

Reply:
Ah, sorry, just saw your response about namespaces... didn't scroll down enough lol :)

Reply:
We could even add direct links to the WebIDL spec using markdown.